src/python/ensembl/io/genomio/fasta/compare.py

```diff
@@ -0,0 +1,284 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Compares the INSDC and core fasta files based on md5sum and ids."""
+
+import argparse
+import logging
+from pathlib import Path
+import re
+from typing import Any, Dict, List, Tuple
+
+from Bio import SeqIO
+
+import ensembl.io.genomio
+from ensembl.utils.archive import open_gz_file
+from ensembl.utils.argparse import ArgumentParser
+from ensembl.utils.logging import init_logging_with_args
+
+__all__ = [
+    "SeqGroup",
+    "CompareFasta",
+]
+
+
+class SeqGroup:
+    """Represents a group of sequence identifiers and maintains a count of them."""
+
+    def __init__(self, identifier: str | None = None) -> None:
+        """
+        Initializes a SeqGroup instance.
+
+        Args:
+            identifier (str | None, optional): The first identifier to add to the group.
+                                               Defaults to None, which adds "None" as the identifier.
+        """
+        self.ids: List[str] = []
+        if identifier:
+            self.add_id(identifier)
+        else:
+            self.add_id("None")
+        self.count = len(self.ids)
+
+    def __str__(self) -> str:
+        """
+        Returns a comma-separated string of sequence identifiers.
+
+        Returns:
+            str: A string representation of the sequence group.
+        """
+        return ", ".join(self.ids)
+
+    def add_id(self, identifier: str | None = None) -> None:
+        """
+        Adds a new identifier to the group and updates the count.
+
+        Args:
+            identifier (str | None, optional): The identifier to add. If None, "None" is added instead.
+        """
+        self.ids.append(identifier if identifier else "None")
+        self.count = len(self.ids)
+
+
+class CompareFasta:
+    """Read and compare the fasta sequences"""
+
+    def __init__(self, fasta1: Path, fasta2: Path, output_dir: str) -> None:
+        self.fasta1 = Path(fasta1)
+        self.fasta2 = Path(fasta2)
+        self.output_dir = Path(output_dir)
+        self.comp: List[str] = []
+
+    def compare_seqs(self) -> None:
+        """
+        Compare two FASTA files for common, unique, and differing sequences.
+        """
+        seq1 = self.read_fasta(self.fasta1)
+        seq2 = self.read_fasta(self.fasta2)
+
+        # Compare sequences
+        seq1_dict = self.build_seq_dict(seq1)
+        seq2_dict = self.build_seq_dict(seq2)
+
+        # Compare number of sequences
+        if len(seq1) != len(seq2):
+            self.comp.append(
+                "WARNING: Different number of sequences: "
+                f"fasta1 [ n = {len(seq1)} ] -Vs- fasta2 [ n = {len(seq2)} ]"
+            )
+            logging.warning("Different number of sequences: fasta1 compared to fasta2")
+
+        common, group_comp = self.find_common_groups(seq1_dict, seq2_dict)
+        self.comp += group_comp
```

Comment:
```suggestion
```
`group_comp` is already `self.comp` (you may want to change the function to only return `common`).

Reply:
yes makes sense, thank you